### ISSUE TITLE
Callable and Concatenate treat non-constraining ParamSpec and literal ellipses differently

### ISSUE THREAD

**Issue by robertmaxton42** [microsoft/pyright#5899]
**Describe the bug**
The two functions 

```
def higher1(f: Callable[Concatenate[int, P], int], *args, **kwargs) -> int:
    return f(*args, **kwargs)

def higher2(f: Callable[Concatenate[int, ...], int], *args, **kwargs) -> int:
    return f(*args, **kwargs)
```
are judged differently by Pyright: the first has an error under `args` (Expected 1 positional argument), the second does not.

I would generally expect the two to behave identically, as `P` is being used in an "unbound" way that does not actually impose any constraints. (In that regard I'm a little surprised not to see a warning under `P` as well.) 

While I think the first judgement is more useful (since presumably we had the intent to ensure that `f` had a parameter list that began with `int`, and an unconstrained `args` does not guarantee that that constraint is satisfied), either way I'd expect the judgement to be consistent.

**VS Code extension or command-line**
Pyright under Pylance VSCode extension, v 2023.8.51.

**Comment by erictraut** [comment]
I agree this is inconsistent. Pyright should either generate an error for both of these cases or neither.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.328, which I just published. It will also be included in a future release of pylance.
